apps/nextjs-website/src/app/auth/sign-up/page.tsx

```diff
@@ -0,0 +1,132 @@
+'use client';
+import { translations } from '@/_contents/translations';
+import PageNotFound from '@/app/not-found';
+import CheckItem from '@/components/molecules/CheckItem/CheckItem';
+import ConfirmSignUp from '@/components/organisms/Auth/ConfirmSignUp';
+import SignUpForm from '@/components/organisms/Auth/SignUpForm';
+import { environment } from '@/config';
+import { SignUpSteps } from '@/lib/types/signUpSteps';
+import { Box, Grid, Typography, useMediaQuery } from '@mui/material';
+import { Auth } from 'aws-amplify';
+import { useCallback, useState } from 'react';
+
+const SignUp = () => {
+  const {
+    auth: { signUp },
+  } = translations;
+
+  const isSmallScreen = useMediaQuery('(max-width: 1000px)');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [company, setCompany] = useState('');
+  const [role, setRole] = useState('');
+  const [signUpStep, setSignUpStep] = useState(SignUpSteps.SIGN_UP);
+
+  const onSignUp = useCallback(async () => {
+    // TODO: Add company and role to user attributes
+    const result = await Auth.signUp({
```

Comment:
If you know fp-ts you can simplify the errors handling and types using the [Either](https://gcanti.github.io/fp-ts/modules/Either.ts.html) structure

Reply:
I'd change with a more functional behaviour when we'll create the custom `AuthProvider`, what do you think?